scripts/gb-model/create_region_shapes.py

```
@@ -0,0 +1,421 @@
+            regions.append(region_data)
+            region_id += 1
+
+    # Create GeoDataFrame from regions
+    if regions:
+        regions_gdf = gpd.GeoDataFrame(regions, crs=country_shapes.crs)
+        logger.info(
+            f"Successfully created {len(regions_gdf)} regions from {len(country_shapes)} original shapes"
+        )
+        return regions_gdf
+    else:
+        logger.error("No regions were created!")
+        return country_shapes.copy()
+
+
+def clean_regions(regions_gdf, min_area_threshold=1000):
+    """
+    Clean up regions by removing very small polygons and fixing invalid geometries.
+
+    Args:
+        regions_gdf (geopandas.GeoDataFrame): Regions to clean
+        min_area_threshold (float): Minimum area threshold for keeping regions
+
+    Returns:
+        geopandas.GeoDataFrame: Cleaned regions
+    """
+    logger.info("Cleaning regions...")
+
+    initial_count = len(regions_gdf)
+
+    # Fix invalid geometries
+    regions_gdf["geometry"] = regions_gdf["geometry"].buffer(0)
+
+    # Calculate areas
+    regions_gdf["area"] = regions_gdf.geometry.area
+
+    # Remove very small regions
+    regions_gdf = regions_gdf[regions_gdf["area"] > min_area_threshold]
+
+    # Reset index
+    regions_gdf = regions_gdf.reset_index(drop=True)
+
+    logger.info(
+        f"Cleaned regions: {initial_count} -> {len(regions_gdf)} (removed {initial_count - len(regions_gdf)} small regions)"
+    )
+
+    return regions_gdf
+
+
+def filter_regions_with_powerplants(regions_gdf, powerplants_path):
+    """
+    Filter regions to keep only those that contain powerplants.
+    
+    Args:
+        regions_gdf (geopandas.GeoDataFrame): Regions to filter
+        powerplants_path (str): Path to powerplants CSV file
+        
+    Returns:
+        geopandas.GeoDataFrame: Filtered regions containing powerplants
+    """
+    logger.info(f"Loading powerplants from: {powerplants_path}")
+    
+    try:
```

remove, rather let the workflow fail with a clear error message